packages/truffle-migrate/index.js

```diff
@@ -224,6 +224,19 @@ var Migrate = {
 
           callback(err, result);
         });
+      },
+      getAddress: function() {
+        var result = "";
+
+        if (provider.getAddress === undefined) return result;
+
+        if (arguments.length === 1) {
+          result = provider.getAddress(arguments[0]);
+        } else {
+          result = provider.getAddress();
+        }
+
+        return result;
```

Comment:
Can't we just simplify this?

Something like:
```javascript
getAddress: function() {
  return provider.getAddress.apply(provider, arguments);
}
```

Not sure if we need the additional undefined check, since I think we want an exception if the underlying provider doesn't have `.getAddress()`. And the arguments can just be forwarded en-masse via `.apply`

Reply:
@gnidan `apply` throw error if `getAddress` function is undefined: Uncaught TypeError: Cannot read property 'apply' of undefined.

Maybe we can change like:

```JS
getAddress: function() {
  if (provider.getAddress === undefined) return '';
  return provider.getAddress.apply(provider, arguments);
}
```

Reply:
Don't we want an exception to happen, though? Or is `''` a reasonable return value in this situation? My thinking is that if the user does `truffleProvider.getAddress` but `provider.getAddress` does not exist, then there should be an exception.